DependencyInjection/LiipHyphenatorExtension.php

```diff
@@ -18,20 +27,52 @@ public function load(array $configs, ContainerBuilder $container)
         $configuration = new Configuration();
         $config = $this->processConfiguration($configuration, $configs);
 
-        foreach ($config as $key => $value) {
-            if ('tokenizers' !== $key && 'filters' !== $key) {
-                $container->setParameter(sprintf('liip_hyphenator.%s', $key), $value);
-            }
-        }
-
-        $options = $container->getDefinition($this->getAlias().'.hyphenator');
+        $options = $container->getDefinition($this->getAlias().'.options');
+        $factory = $container->getDefinition($this->getAlias().'.hyphenator_factory');
 
-        foreach ($config['tokenizers'] as $value) {
-            $options->addMethodCall('addTokenizer', array(new Reference($value)));
-        }
+        foreach ($config as $key => $value) {
+            switch ($key) {
+                case 'hyphen':
+                    $options->addMethodCall('setHyphen', array($value));
+                    break;
+                case 'left_min':
+                    $options->addMethodCall('setLeftMin', array($value));
+                    break;
+                case 'right_min':
+                    $options->addMethodCall('setRightMin', array($value));
+                    break;
+                case 'word_min':
+                    $options->addMethodCall('setWordMin', array($value));
+                    break;
+                case 'quality':
+                    $options->addMethodCall('setQuality', array($value));
+                    break;
+                case 'no_hyphenate_string':
+                    $options->addMethodCall('setNoHyphenateString', array($value));
+                    break;
+                case 'custom_hyphen':
+                    $options->addMethodCall('setCustomHyphen', array($value));
+                    break;
 
-        foreach ($config['filters'] as $value) {
-            $options->addMethodCall('addFilter', array(new Reference($value)));
+                case 'tokenizers':
```

Comment:
if https://github.com/heiglandreas/Org_Heigl_Hyphenator/pull/21 gets merged we can set the tokenizers and filter instances on the Options